build-system/tasks/helpers.js

```diff
@@ -415,7 +415,7 @@ async function esbuildCompile(srcDir, srcFilename, destDir, options) {
    */
   function remapDependenciesPlugin() {
     const remaps = Object.entries(options.remapDependencies).map(
-      ([path, value]) => ({regex: new RegExp(`^${path}$`), value})
+      ([path, value]) => ({regex: new RegExp(`^${path}(\.[jt]sx?)?$`), value})
```

Comment:
nit: unwrap regex parts for readability and discoverability

```suggestion
      ([path, value]) => ({regex: new RegExp(`^${path}(\.js|\.jsx|\.ts|\.tsx)?$`), value})
```